### PR TITLE
LPD-15146 Adapt headless methods with new fields.

### PR DESCRIPTION
**Pull request by manuelRives**
# What is this trying to solve?

-https://liferay.atlassian.net/browse/LPD-15146 Adapt headless methods with new fields

# How am I fixing it?

Modifying the document API definition to incorporate these new fields.

# How can you verify that it works?
  1. Create a content assigning it a publication date and an expiration date.
  2. Retrieve the content using the API GUI.
  3. Check that the new fields are populated.

### CODE REVIEW

**Comment by AliciaGarciaGarcia** [comment]
ci:tesT:sf

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

File: modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/dto/v1_0/converter/DocumentDTOConverter.java [hunk 1]
```diff
@@ -155,7 +155,9 @@ public Document toDTO(DTOConverterContext dtoConverterContext)
 						fileEntry.getCompanyId(),
 						dtoConverterContext.getLocale()));
 				setDateCreated(fileEntry::getCreateDate);
+				setDateExpired(fileEntry::getExpirationDate);
```
Comment (adolfopa) [inline modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/dto/v1_0/converter/DocumentDTOConverter.java:159]: Q: shouldn't these be in the same order as in the yaml declaration?

**Comment by manuelRives** [comment]
ci:test:sf

File: modules/apps/headless/headless-delivery/headless-delivery-test/src/testIntegration/java/com/liferay/headless/delivery/resource/v1_0/test/DocumentResourceTest.java [hunk 2]
```diff
@@ -238,11 +252,1003 @@ protected void assertValid(
 			_read("http://localhost:8080" + document.getContentUrl()));
 	}
 
+	@Override
+	protected boolean equals(Document document1, Document document2) {
+		if (document1 == document2) {
```
Comment (AliciaGarciaGarcia) [inline modules/apps/headless/headless-delivery/headless-delivery-test/src/testIntegration/java/com/liferay/headless/delivery/resource/v1_0/test/DocumentResourceTest.java:257]: Idea:

```
	@Override
	protected boolean equals(Document document1, Document document2) {
if (!FeatureFlagManagerUtil.isEnabled(
					testCompany.getCompanyId(), "LPD-10701") ){
return super.equals();
}
```

and then this code without the FFs


This way we can be sure on the removal that this code should be the reemplace once the removal of the FF

**Pull request closed without merging**